src/lib/mergeThemes.ts

```diff
@@ -0,0 +1,124 @@
+const mergeComponentVariables = (
+  target: IThemeComponentVariablesInput,
+  ...sources: IThemeComponentVariablesInput[]
+): IThemeComponentVariablesPrepared => {
+  const displayNames = _.union(_.keys(target), ..._.map(sources, _.keys))
+
+  return sources.reduce((acc, next) => {
+    return displayNames.reduce((componentVariables, displayName) => {
+      // Break references to avoid an infinite loop.
+      // We are replacing functions with new ones that calls the originals.
+      const originalTarget = acc[displayName]
+      const originalSource = next[displayName]
+
+      componentVariables[displayName] = siteVariables => {
+        return {
+          ...callable(originalTarget)(siteVariables),
+          ...callable(originalSource)(siteVariables),
+        }
+      }
+
+      return componentVariables
+    }, {})
+  }, target)
+}
+
+/**
+ * See mergeComponentVariables() description.
+ * Component styles adhere to the same pattern as component variables, except
+ *   that they return style objects.
+ */
+const mergeComponentStyles = (
+  target: IThemeComponentStylesInput,
+  ...sources: IThemeComponentStylesInput[]
+): IThemeComponentStylesPrepared => {
+  const initial: IThemeComponentStylesPrepared = _.mapValues(target, stylesByPart => {
+    return _.mapValues(stylesByPart, callable)
+  })
+
+  return sources.reduce<IThemeComponentStylesPrepared>((acc, next) => {
+    _.forEach(next, (stylesByPart, displayName) => {
+      acc[displayName] = acc[displayName] || {}
+
+      _.forEach(stylesByPart, (partStyle, partName) => {
+        // Break references to avoid an infinite loop.
+        // We are replacing functions with a new ones that calls the originals.
+        const originalTarget = acc[displayName][partName]
+        const originalSource = next[displayName][partName]
+
+        acc[displayName][partName] = styleParam => {
+          return _.merge(callable(originalTarget)(styleParam), callable(originalSource)(styleParam))
+        }
+      })
+    })
+
+    return acc
+  }, initial)
+}
+
+const mergeRTL = (target, ...sources) => {
+  return !!sources.reduce((acc, next) => {
+    return typeof next === 'boolean' ? next : acc
+  }, target)
+}
+
+const mergeThemes = (...themes: IThemeInput[]): IThemePrepared => {
+  const emptyTheme = {
+    siteVariables: {},
+    componentVariables: {},
+    componentStyles: {},
+  } as IThemePrepared
+
+  return themes.reduce<IThemePrepared>((acc: IThemePrepared, next: IThemeInput) => {
+    if (!next) return acc
+
+    acc.siteVariables = mergeSiteVariables(acc.siteVariables, next.siteVariables)
+
+    acc.componentVariables = mergeComponentVariables(
+      acc.componentVariables,
+      next.componentVariables,
+    )
+
+    acc.componentStyles = mergeComponentStyles(acc.componentStyles, next.componentStyles)
+
+    // Latest RTL value wins
+    acc.rtl = mergeRTL(acc.rtl, next.rtl)
+
+    // Use the correct renderer for RTL
+    acc.renderer = acc.rtl ? felaRtlRenderer : felaRenderer
```

Comment:
Should `emptyTheme` have a default renderer? It looks to be possible to never set `renderer` if `next` is never truthy. Same goes for `rtl`.